lib/User/Manager.php

```diff
@@ -267,42 +267,32 @@ public function getCachedEntry($uid) {
 	}
 
 	/**
-	 * checks whether a user is still available on LDAP
+	 * Try to resolve missing DN by UUID
 	 *
 	 * @param string $dn
 	 * @return bool
-	 * @throws \Exception
-	 * @throws \OC\ServerNotAvailableException
 	 */
 	public function dnExistsOnLDAP($dn) {
-
-		//check if user really still exists by reading its entry
-		if (!\is_array($this->access->readAttribute($dn, '', $this->getConnection()->ldapUserFilter))) {
-			$lcr = $this->getConnection()->getConnectionResource();
-			if ($lcr === null) {
-				throw new \Exception('No LDAP Connection to server ' . $this->getConnection()->ldapHost);
-			}
-
-			try {
-				$uuid = $this->access->getUserMapper()->getUUIDByDN($dn);
-				if (!$uuid) {
-					return false;
-				}
+		try {
+			$uuid = $this->access->getUserMapper()->getUUIDByDN($dn);
+			if ($uuid) {
 				$newDn = $this->access->getUserDnByUuid($uuid);
 				//check if renamed user is still valid by reapplying the ldap filter
 				if (!\is_array($this->access->readAttribute($newDn, '', $this->getConnection()->ldapUserFilter))) {
 					return false;
 				}
 				$this->access->getUserMapper()->setDNbyUUID($newDn, $uuid);
 				return true;
-			} catch (\Exception $e) {
-				$this->logger->logException($e, ['app' => self::class]);
-				return false;
 			}
+		} catch (ServerNotAvailableException $e) {
+			// A. pass? Acc will deleted then.
+			// B. fail? in the middle of the user sync?
```

Comment:
I don't think we can handle that scenario here... What could happen if we let it bubble up? Any other place where we could handle that scenario?

Reply:
ServerNotAvailableException is propagated to the Symfony application and user sync is aborted:
```
> php occ user:sync -m remove "OCA\User_LDAP\User_Proxy"
Analysing known accounts ...
   27 [--------------------------->]
In Manager.php line 278:
        
  Boom  
        

user:sync [-l|--list] [-u|--uid UID] [-s|--seenOnly] [-c|--showCount] [-m|--missing-account-action MISSING-ACCOUNT-ACTION] [-r|--re-enable] [--] [<backend-class>]
```
This is bad according to @cdamken as user sync is done in background and nobody will notice this.

Reply:
From my point of view, I think the admins should wrap the command in a way it suit their needs. They can check that the command has crashed if it ends with a non-0 status code (I guess we should be returning 1 now), so in this situation they can send a email using any command-line tool so they're notified in order to take any action.

I think the command must crash. There is nothing we can do to fix this situation. The only question that remains is if we should send somehow an email to the admin group, but this assumes at least one of the admins has set a valid email and that such admin can fix the problem (maybe the admin can't access to the host but just to ownCloud).
Note that the sync service will also increase the complexity. There is already a lot of logic there. Adding more logic implies a higher probability of bugs happening around something quite important as syncing the users.

Reply:
>we should be returning 1 now

exactly:
```
> echo $?
1
```
Updated to crash on `ServerNotAvailableException`